src/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -429,6 +429,80 @@ public static bool TrySkipBlock(BinaryReader reader)
 
             return true;
         }
+
+        public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
+        {
+            const int OffsetToFilenameLength = 22; // from the point after the signature
+
+            if (reader.ReadUInt32() != SignatureConstant)
+                return false;
+
+            if (reader.BaseStream.Length < reader.BaseStream.Position + OffsetToFilenameLength)
+                return false;
+
+            uint minimunVersion = reader.ReadUInt16();
+            int dataDescriptorBit = reader.ReadUInt16() & 0x08;
+            reader.BaseStream.Seek(10, SeekOrigin.Current);
+            long compressedSize = reader.ReadUInt32();
+            long uncompressedSize = reader.ReadUInt32();
+            int filenameLength = reader.ReadUInt16();
+            uint extraFieldLength = reader.ReadUInt16();
+            reader.BaseStream.Seek(filenameLength, SeekOrigin.Current);
+            long endExtraFields = reader.BaseStream.Position + extraFieldLength;
+            
+            if (dataDescriptorBit == 0)
+            {
+                bool uncompressedSizeInZip64 = uncompressedSize == ZipHelper.Mask32Bit;
```

Comment:
nit: consider prefixing these with `is` to make it clearer that they are bools and not sizes.